src/announcements/AnnouncementsContext.tsx

```diff
@@ -29,10 +42,23 @@ const AnnouncementsContextProvider: React.FC = ({children}) => {
     return () => unsubscribe();
   }, []);
 
+  const dismissAnnouncement = (dismissedAnnouncement: AnnouncementType) => {
+    addDismissedAnnouncementInStore(dismissedAnnouncement.id);
+    setAnnouncements((announcements) =>
+      announcements.filter((a) => a.id !== dismissedAnnouncement.id),
+    );
+  };
+
+  const resetDismissedAnnouncements = () => {
+    setDismissedAnnouncementInStore([]);
+  };
```

Comment:
Wrap these with `useCallback`, just in case. When we return them from a Context, we kinda "loose control" of how they are used, and the function hash should be stable so using them in other hooks don't cause rerenders.